src/pages/welcome.js

```diff
@@ -0,0 +1,239 @@
+import React from 'react';
+import {Router} from '@reach/router';
+import { ApolloClient } from 'apollo-client';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { graphql } from 'gatsby';
+import { httpLink } from '../utils/apollo';
+import gql from 'graphql-tag';
+import {has} from 'lodash';
+
+
+import {Loading} from '../components/organims/Loading';
+import DownloadByOrganization from '../components/organims/DownloadByOrganization';
+import HasBeenDownloaded from '../components/organims/HasBeenDownloaded';
+
+
+const getOrgInfo = gql`
+  query($slug: String!) {
+    organizationBySlug(slug: $slug) {
+      slug,
+      name,
+      domain,
+      pictureUrl,
+    }
+  }
+`;
+
+export default class Welcome extends React.Component{
+  constructor(props){
+    super(props);
+    this.client = new ApolloClient({
+      link: httpLink,
+      cache: new InMemoryCache(),
+    });
+    this.state={
+      step: 1,
+      isLoading: true,
+      unreachable: false,
+      filledOrgData:{
+        title: 'Your organization name',
+        description: 'Your organization description',
+        details: 'Your organization details',
+        email: 'name.org',
+        pictureUrl: '',
+      },
+    }
+  }
+  handleClickDownload(){
+    const{data} = this.props.data.prismicWelcome;
+    window.open(data.button_url.url);
+    this.setState({
+      step: 2
+    })
+  }
+  componentDidMount(){
+    const {data} = this.props.data.prismicWelcome;
+    const {client} = this;
+    // this is sent by gatzby matcher
+    const matchedUrl =  this.props['*'];
+    const splittedPathName = matchedUrl.split('/');
+    const organizationSlug = splittedPathName.length > 0 ? splittedPathName[1] : null;
+    // We do not want netlify to do this..
+    if(!window) return 
+    if(!organizationSlug){
+      this.setState({
+        isLoading: false,
+        unreachable: true,
+      });
+      return;
+    }
+    const currentUrl = new URL(window.location);
+    if(currentUrl.search && currentUrl.search.includes('?dl')){
+      this.setState({
+        step: 2,
+      })
+      window.location = data.button_url.url;
+    }
+
+    // Get data to inject from our API
+    client
+    .query({
+      query: getOrgInfo,
+      variables: { slug: organizationSlug}
+    })
+    .then(res=>{
+      // Insufisent data, go to 404
+      const propsToChecks = 'data.organizationBySlug';
+      if(!has(res, propsToChecks) || (!has(res, `${propsToChecks}.name`) && !has(res, `${propsToChecks}.domain`))) {
```

Comment:
Hum in fact it's the case where there are *no name AND no domain* (in fact no data about this organization go there)

Reply:
OK, indeed! i checked it was working properly by removing my organization's name.

Note that the case where `organizationBySlug` is not null and `organizationBySlug.domain` can not happen because of schema's restriction.